File: src/main/java/uk/gov/justice/digital/hmpps/riskprofiler/model/IncidentParty.java
package uk.gov.justice.digital.hmpps.riskprofiler.model;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.*;

import java.io.Serializable;

@JsonInclude(JsonInclude.Include.NON_NULL)
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class IncidentParty implements Serializable {

    private static final long serialVersionUID = 1L;

    private Long bookingId;
    private Long partySeq;
    private Long staffId;
    private Long personId;
    private String participationRole;
    private String outcomeCode;
    private String commentText;
    private Long incidentCaseId;
}

File: src/main/kotlin/uk/gov/justice/digital/hmpps/riskprofiler/model/IncidentParty.kt
package uk.gov.justice.digital.hmpps.riskprofiler.model

import com.fasterxml.jackson.annotation.JsonInclude
import java.io.Serializable

@JsonInclude(JsonInclude.Include.NON_NULL)
data class IncidentParty(
  val bookingId: Long? = null,
  val partySeq: Long? = null,
  val staffId: Long? = null,
  val personId: Long? = null,
  val participationRole: String? = null,
  val outcomeCode: String? = null,
  val commentText: String? = null,
  val incidentCaseId: Long? = null,
) : Serializable
